newWith/src/macro.scala

```diff
@@ -0,0 +1,39 @@
+import scala.quoted.*
+import scala.quoted.runtime.impl.QuotesImpl
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.core.Contexts.Context
+
+inline def mcr[T]: Base = ${ mcrImpl[T] }
+def mcrImpl[T: Type](using Quotes): Expr[Base] =
+  import quotes.reflect.*
+  given ctx: Context = quotes.asInstanceOf[QuotesImpl].ctx
+
+  class MyTreeMap extends TreeMap:
+    override def transformStatement(tree: Statement)(owner: Symbol): Statement =
+      println(s"Traversing tree ${tree.getClass}")
+      tree match
+        case t: tpd.TypeDef =>  // matching against quotes.reflect.TypeDef doesn't work
```

Comment:
```suggestion
        case t: tpd.ClassDef =>
```

Reply:
Did you try `tree.show(using Printer.TreeStructure)`?

Reply:
Hmm, it produces different output than just `println(tree)`. What's the difference between the two?